project/MimaExcludes.scala

```diff
@@ -66,6 +66,12 @@ object MimaExcludes {
     ProblemFilters.exclude[Problem]("org.sparkproject.spark_core.protobuf.*"),
     ProblemFilters.exclude[Problem]("org.apache.spark.status.protobuf.StoreTypes*"),
 
+    // SPARK-43265: Move Error framework to a common utils module
+    ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.QueryContext"),
+    ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.SparkException"),
+    ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.SparkException$"),
+    ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.SparkThrowable"),
```

Comment:
Did we rename the package? Otherwise it should be still binary compatible.

Reply:
I don't think I renamed the package name. 

This throws from spark-core? Unless spark-core builds the iber jar and this checks the uber jar otherwise it should throw?

Reply:
Can we spend a bit more time on this? We are going to move quite a few classes, and it would be great if MiMa works to avoid mistakes.

Reply:
First all I am thinking existing Mima check just compile the `core` and compare it with previous 3.4.0 so we are moving out classes thus seeing the report is expected.

If we want to do a step further to say even after moving the classes, we can still check the compatibility between core 3.4.0 and `core+common-utils`, probably we can follow what `CheckConnectJvmClientCompatibility` does:

1. Pre-run a command to compile a jar that include both `core` and `common-util` (not sure if SBT can do this).
2. The new suite locates the jar from 1., and then also find the 3.4.0 jar.
3. Then the suite compare two jars.